web/handlers.go

```diff
@@ -177,15 +178,52 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("X-Frame-Options", "SAMEORIGIN")
 
 		// Add unsafe-eval to the content security policy for faster source maps in development mode
-		devCSP := ""
+		devCSPMap := make(map[string]bool)
 		if model.BuildNumber == "dev" {
-			devCSP += " 'unsafe-eval'"
+			devCSPMap["unsafe-eval"] = true
 		}
 
 		// Add unsafe-inline to unlock extensions like React & Redux DevTools in Firefox
 		// see https://github.com/reduxjs/redux-devtools/issues/380
 		if model.BuildNumber == "dev" {
-			devCSP += " 'unsafe-inline'"
+			devCSPMap["unsafe-inline"] = true
+		}
+
+		// Add supported flags for debugging during development, even if not on a dev build.
+		for _, devFlagKeyValueStr := range strings.Split(*c.App.Config().ServiceSettings.DeveloperFlags, ",") {
+			devFlagKeyValueSplit := strings.SplitN(devFlagKeyValueStr, "=", 2)
+			if len(devFlagKeyValueSplit) != 2 {
+				c.Logger.Warn("Unable to parse developer flag", mlog.String("developer_flag", devFlagKeyValueStr))
+				continue
+			}
+			devFlagKey := devFlagKeyValueSplit[0]
+			devFlagValue := devFlagKeyValueSplit[1]
+
+			// Ignore disabled keys
+			if devFlagValue != "true" {
+				continue
+			}
+
+			// Honour only supported keys
+			switch devFlagKey {
+			case "unsafe-eval", "unsafe-inline":
+				devCSPMap[devFlagKey] = true
+			default:
+				c.Logger.Warn("Unrecognized developer flag", mlog.String("developer_flag", devFlagKeyValueStr))
+			}
+		}
+
+		// Build a slice for sorting
+		var devCSPKeys []string
+		for key := range devCSPMap {
+			devCSPKeys = append(devCSPKeys, key)
+		}
+		sort.Strings(devCSPKeys)
```

Comment:
Wondering why do we need to sort the keys? I don't believe they are order sensitive.

Reply:
The order here is necessary to ensure a stable CSP header directive. This is necessary at least for tests, though I'm not sure it has an impact elsewhere.

Reply:
Indeed. In that case, I'd just suggest to check for key existence in the map and then append that in the string.